libgstc/pygstc/gstcerror.py

```diff
@@ -28,25 +28,118 @@
 # ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
 # OF THE POSSIBILITY OF SUCH DAMAGE.
 
+from enum import Enum
+
+"""
+GSTC - GstClientError Class
+Main GstClientError Class for the GstClient
+It groups together all the GstClient Errors
+"""
+
+
+class GstClientError(Exception):
+
+    def __init__(self, description, code):
+        """
+        Raised when GstClient fails
+
+        Parameters
+        ----------
+        Attributes:
+            description -- GstClient error description message
+        """
+        self.description = description
+        self.code = code
+
+
 """
 GSTC - GstdError Class
+Raised when Gstd IPC fails
 """
 
 
-class GstdError(Exception):
+class GstdError(GstClientError):
 
-    """Raised when Gstd IPC fails"""
+    def __init__(self, description, code):
+        """
+        Initialize new GstdError
 
-    pass
+        Parameters
+        ----------
+        Attributes:
+            description : GstdError description message
+            code : Error code
+        """
+        super(GstdError, self).__init__(description, code)
```

Comment:
This is the name of the parent class

Reply:
I think it should be: _super(BaseClass..._

Reply:
Right. I don't think you need the arguments.